nemo/collections/nlp/models/text_classification/text_classification_model.py

```diff
@@ -200,6 +190,14 @@ def setup_training_data(self, train_data_config: Optional[DictConfig]):
             return
         self._train_dl = self._setup_dataloader_from_config(cfg=train_data_config)
 
+        # we need to create/update the loss module by using the weights calculated from the training data
+        if self.cfg.dataset.class_balancing == 'weighted_loss':
+            self.class_weights = calc_class_weights(train_data_config.file_path, self.cfg.dataset.num_classes)
+            self.loss = CrossEntropyLoss(weight=self.class_weights)
+        else:
+            self.class_weights = None
+            self.loss = CrossEntropyLoss()
+
```

Comment:
I would rather have this in a separate method for the users to call but this would require changes on the TLT side.

Reply:
Moved it into a method.